### PR TITLE
Validate the format configuration

### PR DESCRIPTION
**Pull request by tijmenb**
These tests validate that our format config is internally consistent and all the required fields have been set.

Also fixes 2 mistakes in the current config from https://github.com/alphagov/content-publisher/pull/26: `detailed_guides ` was written plural, and `publication` is not a document type (it's a schema).

https://trello.com/c/egKyMGRP

### CODE REVIEW

File: spec/formats/configuration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe "Format configuration" do
+  supertypes = YAML.load_file("app/formats/supertypes.yml")
+  document_types = YAML.load_file("app/formats/document_types.yml")
+
+  describe "Supertypes" do
```
Comment (benthorner) [inline spec/formats/configuration_spec.rb]: Could we collapse these two describes into one to make it a bit more readable?

Comment (benthorner) [reply spec/formats/configuration_spec.rb]: (Or the inner describe and the it - it's very block-tastic at the moment ;-).

File: spec/formats/configuration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe "Format configuration" do
+  supertypes = YAML.load_file("app/formats/supertypes.yml")
```
Comment (benthorner) [inline spec/formats/configuration_spec.rb]: Would a 'let' be more conventional here?

**Comment by benthorner** [comment]
With the format files, I'm confused as to why we've put these config files in the app/ directory - usually config files go in config/ and get loaded into the environment so we don't have to keep doing 'YAML.load' everywhere. Could we look at doing that?

File: spec/formats/configuration_spec.rb (hunk 1 above)
Comment (tijmenb) [reply spec/formats/configuration_spec.rb]: Good point, I merged the two `describes`!

File: spec/formats/configuration_spec.rb (hunk 2 above)
Comment (tijmenb) [reply spec/formats/configuration_spec.rb]: As it turns out, you can only use `let` inside an `it` block (there are other problems with `let` but we'll talk about those later :)).

**Comment by tijmenb** [comment]
> With the format files, I'm confused as to why we've put these config files in the app/ directory - usually config files go in config/ and get loaded into the environment so we don't have to keep doing 'YAML.load' everywhere. Could we look at doing that?

@benthorner I would have thought you'd seen it in https://github.com/alphagov/content-publisher/pull/26/commits/7b1d315af7d57943a0b9cd30510bdde3d4d26da0 😄 

> Added the YAML files to `/app/formats` which may be controversial. Not sure where to put them though - they're so core to the app that `config` feels wrong.

and

I'm expecting follow up PRs to:

- Refactor YAML handling. I've resisted the temptation to refactor the `YAML.load_file` stuff into classes at this time, because I want to avoid making the decisions about naming / placing stuff now. For what it's worth, I'm thinking about something like `DocumentType.all` and `DocumentType.find("press_release")`.

Basically, the YAML files feel more business logic than config.

How do you get to the contents of a file from the contents?

File: spec/formats/configuration_spec.rb (hunk 2 above)
Comment (tijmenb) [reply spec/formats/configuration_spec.rb]: Clarification: you can only use `let` in a `describe` block, and the resulting variable inside an `it` block. Otherwise it will raise an error like:

> `foo` is not available on an example group (e.g. a `describe` or `context` block). It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
